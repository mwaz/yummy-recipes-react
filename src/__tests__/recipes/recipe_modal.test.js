import React from 'react';
import { shallow, mount, render } from 'enzyme';
import RecipeModal from '../../components/common/recipe-modal';
import toJson, { shallowToJson } from 'enzyme-to-json';

describe('RecipeModal component', () => {
    const recipes= jest.fn();
    const wrapper = shallow(<RecipeModal recipes={recipes} location={{}} />);
    const preventDefault = jest.fn();

    it('renders properly without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});