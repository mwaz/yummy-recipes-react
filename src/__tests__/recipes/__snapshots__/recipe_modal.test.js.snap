// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeModal component renders properly without crashing 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  className="modal-fade"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <form>
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
         
         
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        className="form-group"
      >
        <label
          className="col-md-6 control-label"
          htmlFor="recipe_name"
        >
          Recipe Name
        </label>
        <div
          className="col-md-12"
        >
          <input
            className="form-control"
            id="recipe_name"
            type="text"
          />
        </div>
        <label
          className="col-md-6 control-label"
          htmlFor="recipe_ingredients"
        >
          Recipe Ingredients
        </label>
        <div
          className="col-md-12"
        >
          <textarea
            className="form-control"
            cols="50"
            id="recipe_ingredients"
            rows="4"
          />
        </div>
        <label
          className="col-md-6 control-label"
          htmlFor="recipe_methods"
        >
          Recipe Methods
        </label>
        <div
          className="col-md-12"
        >
          <textarea
            className="form-control"
            cols="50"
            id="recipe_methods"
            rows="4"
          />
        </div>
      </div>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="success"
        disabled={false}
        id="recipe_modal_button"
        type="submit"
      >
         
         
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="info"
        disabled={false}
        id="cancel"
      >
         Cancel
      </Button>
    </ModalFooter>
  </form>
</Modal>
`;
